### PR TITLE
[WasmFS] Fix other.test_realpath_nodefs

### PR DESCRIPTION
**Pull request by kripken**
This is the usual issue with the old JS FS including more JS API code automatically, so
things just happen to work. For WasmFS, doing `FS.mkdir` etc. requires telling emcc
that we want the JS API to exist.

### CODE REVIEW

**Comment by sbc100** [comment]
I the long run that are we going to do about this discrepancy between the old and new filesystems?

**Review by sbc100 — APPROVED** [review]

File: test/test_other.py [hunk 1]
```diff
@@ -7154,7 +7155,7 @@ def test_realpath_nodefs(self):
 }
 ''')
     create_file('TEST_NODEFS.txt', ' ')
-    self.run_process([EMCC, 'src.c', '-lnodefs.js'])
+    self.emcc('src.c', args=['-lnodefs.js', '-sFORCE_FILESYSTEM'])
     self.assertContained('Resolved: /working/TEST_NODEFS.txt', self.run_js('a.out.js'))
```
Comment (sbc100) [inline test/test_other.py:7159]: Can you instead convert both these two lines into a single call to `self.do_runf` ?

**Comment by kripken** [comment]
> I the long run that are we going to do about this discrepancy between the old and new filesystems?

This is unavoidable - we'll just need to document it (I'll have a PR with that soon). The issue is that in the old FS we just get all the JS code all the time anyhow - that JS code is the actual implementation, so it's just there, and DCE doesn't remove it, so user JS can call it. For WasmFS, OTOH, we don't want to force this code to be included all the time - it would be a big code size burden. So the user needs to explicitly say "I want this JS code". There were situations in the old code that also required the user to say that (cases where DCE *did* work :smile: ) but they were just more rare.

File: test/test_other.py [hunk 2]
```diff
@@ -7154,7 +7155,7 @@ def test_realpath_nodefs(self):
 }
 ''')
     create_file('TEST_NODEFS.txt', ' ')
-    self.run_process([EMCC, 'src.c', '-lnodefs.js'])
+    self.emcc('src.c', args=['-lnodefs.js', '-sFORCE_FILESYSTEM'])
```
[the remark this answers is not in the document]
Comment (kripken) [reply test/test_other.py:7158]: Done.

File: test/test_other.py (hunk 1 above)
Comment (kripken) [reply test/test_other.py:7159]: Done.

**Review by tlively — APPROVED** [review]

**Pull request merged**